backend/site-api/src/app/site/get_current_site.rs

```diff
@@ -70,30 +72,60 @@ pub async fn get_current_site(
             .await
             .map_err(|_| ApiError::not_found().message("Site preview not found"))?
     } else {
-        let site = context
-            .site_repo
-            .get_site_latest_version(&site_id, true)
-            .await
-            .map_err(|e| ApiError::not_found().message(e))?;
+        // Get site from cache
+        let json = context.cache.get_usage(&site_id).await?.site_data;
+        let cache_result: Result<SiteEntity, serde_json::Error> = serde_json::from_value(json);
+
+        let site = match cache_result {
+            Ok(site) => site,
+            Err(_) => context
+                .site_repo
+                .get_site_latest_version(&site_id, true)
+                .await
+                .map_err(|e| ApiError::not_found().message(e))?,
+        };
 
-        if !site.published && !is_admin_or_site_owner(&site_metadata, &user) {
+        if !site.published && !is_admin_or_site_owner(&owner_id, &user) {
             return Err(ApiError::bad_request().code(ApiErrorCode::SiteUnpublished));
         }
         site
     };
 
-    let site_size = site.calculate_site_size();
-
     // Check if the bandwidth usage exceeds the allowed limit
     context
         .cache
-        .check_bandwidth_exceeded(&site_id, site_size, site_metadata.site_type)
+        .check_bandwidth_exceeded(&site_id, &site, site_type)
         .await?;
 
     context
         .cache
-        .increase_request_count(&site_id, site_size, site_metadata.site_type)
+        .increase_request_count(&site_id, &site, site_type)
         .await;
 
-    Ok(to_api_response(site))
+    // Only include public pages
+    let filtered_pages = filter_pages(&site.pages)?;
+    println!("Filtered Pages: {}", filtered_pages);
+
+    Ok(Json(to_api_response(&site, filtered_pages)))
+}
+
+fn filter_pages(pages: &str) -> Result<String, ApiError> {
+    let cleaned_pages = pages.trim().trim_matches('"').replace("\\", "");
```

Comment:
I got stuck here for a while because I encountered an error when converting the JSON string of pages into a `Map<String, Value>`. 

Initially, I hardcoded the exact same string and it passed. Then I converted both to bytes to compare them that I realized there might be hidden extra spaces, escape characters, and so on in the string obtained from `site.pages`. Therefore, I added this step to clean up the data.

Reply:
That's because the JSON is "double" stringified! It's wasteful, but I couldn't figure out how to get things working between frontend/backend any other way.

The solution is to parse to `Value` first, and then parse the result as `Map<String, Value>`